tensorboard/java/org/tensorflow/tensorboard/vulcanize/Vulcanize.java

```diff
@@ -614,11 +618,28 @@ private static void rootifyAttribute(Node node, String attribute) {
       return;
     }
     Webpath uri = Webpath.get(value);
-    if (webfiles.containsKey(uri)) {
-      node.attr(attribute, outputPath.getParent().relativize(uri).toString());
+    // Form absolute path from uri if uri is not an absolute path.
+    // Note that webfiles is a map of absolute webpaths to relative filepaths.
+    Webpath absUri = isAbsolutePath(uri)
+        ? uri : me().getParent().resolve(uri).normalize();
+
+    if (webfiles.containsKey(absUri)) {
+      node.attr(attribute, outputPath.getParent().relativize(absUri).toString());
     }
   }
 
+  /**
+   * Checks whether a path is a absolute path.
+   * Webpath.isAbsolute does not take data uri and other forms of absolute path into account.
+   */
+  private static Boolean isAbsolutePath(Webpath path) {
+    String sPath = path.toString();
+    return path.isAbsolute()
+        || sPath.startsWith(DATA_URI_PREFIX)
```

Comment:
Why special-case these schemes, instead of just checking whether the URL
matches `^(?:/|[A-Za-z][A-Za-z0-9+.-]*:)`?

(Ref.: <https://tools.ietf.org/html/rfc3986#appendix-A>)